src/microbe_stage/Patch.cs

```diff
@@ -79,6 +79,35 @@ public bool AddNeighbour(Patch patch)
         return Adjacent.Add(patch);
     }
 
+    /// <summary>
+    ///   Checks all neighbors recursively to find all connected patch nodes
+    /// </summary>
+    /// <returns>A <see cref="HashSet{T}"/> of <see cref="Patch"/> connected to this node by some means</returns>
+    public HashSet<Patch> GetAllConnectedPatches()
+    {
+        HashSet<Patch> resultSet = new HashSet<Patch>();
+        CollectNeighbors(this, ref resultSet);
+
+        return resultSet;
+    }
+
+    /// <summary>
+    ///   Adds all neighbors recursively to the referenced <see cref="HashSet{T}"/>
+    /// </summary>
+    /// <param name="patch">The <see cref="Patch"/> to start from</param>
+    /// <param name="set">The <see cref="HashSet{T}"/> to add to</param>
+    public void CollectNeighbors(Patch patch, ref HashSet<Patch> set)
+    {
+        foreach (var neighbour in patch.Adjacent)
+        {
+            if (!set.Contains(neighbour))
```

Comment:
HashSet.Add already returns a boolean indicating whether the element was added, we can save one method call like that (so instead of first contains and then add, just directly add and check the return value).
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.add?view=net-5.0